manifests/config.pp

```diff
@@ -5,23 +5,23 @@
 {
 
   group { $qpid::group:
-    ensure => present,
+    ensure => $qpid::ensure,
   }
 
   user { $qpid::user:
```

Comment:
Since no ordering is defined here between the user and group, Puppet will use the manifest ordering by default.

That means when running with `ensure => absent`, deleting the group will fail with: `groupdel: cannot remove the primary group of user 'qpidd'`

Changing the order in which these two resources are declared in the manifest resolved this issue for me.